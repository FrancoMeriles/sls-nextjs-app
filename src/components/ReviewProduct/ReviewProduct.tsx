import { FC } from 'react'
import { Tabs, TabList, TabPanels, Tab, TabPanel } from '@chakra-ui/react'

// Components
import ReviewProductItem from '@base/components/ReviewProductItem'
import ReviewProductHeader from '@base/components/ReviewProductHeader'

interface ReviewProductProps {
  allReviews: ReviewInterface[]
  ratingAverage: number
  sumRatingsOpinions: number
  ratingLevels: RatingLevelsInterface
}

interface ReviewInterface {
  id: number
  rate: number
  title: string
  content: string
  likes: number
  dislikes: number
}

interface RatingLevelsInterface {
  five_star: number
  four_star: number
  one_star: number
  three_star: number
  two_star: number
}

const ReviewProduct: FC<ReviewProductProps> = ({
  allReviews,
  ratingAverage,
  sumRatingsOpinions,
  ratingLevels,
}) => {
  const positiveReviews = allReviews.filter((review: ReviewInterface) => review.rate > 3)
  const negativeReviews = allReviews.filter((review: ReviewInterface) => review.rate < 3)
  return ratingAverage ? (
    <>
      <ReviewProductHeader
        ratingAverage={ratingAverage}
        sumRatingsOpinions={sumRatingsOpinions}
        ratingLevels={ratingLevels}
      />
      <Tabs isFitted colorScheme="brand">
        <TabList>
          <Tab>Todas</Tab>
          <Tab>Positivas</Tab>
          <Tab>Negativas</Tab>
        </TabList>

        <TabPanels>
          <TabPanel p={0}>
            {allReviews.map((review: ReviewInterface) => (
              <ReviewProductItem review={review} key={review.id} />
            ))}
          </TabPanel>
          <TabPanel p={0}>
            {positiveReviews.map((review: ReviewInterface) => (
              <ReviewProductItem review={review} key={review.id} />
            ))}
          </TabPanel>
          <TabPanel p={0}>
            {negativeReviews.map((review: ReviewInterface) => (
              <ReviewProductItem review={review} key={review.id} />
            ))}
          </TabPanel>
        </TabPanels>
      </Tabs>
    </>
  ) : null
}

export default ReviewProduct
