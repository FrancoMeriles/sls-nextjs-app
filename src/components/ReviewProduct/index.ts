import ReviewProduct from './ReviewProduct'

export default ReviewProduct
