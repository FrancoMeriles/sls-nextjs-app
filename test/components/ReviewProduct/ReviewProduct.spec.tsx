import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import ReviewProduct from '@base/components/ReviewProduct'

const allReviews = [
  {
    id: 123456,
    rate: 4,
    title: '__TITLE_ONE__',
    content: '__CONTENT_ONE__',
    likes: 3,
    dislikes: 1,
  },
  {
    id: 1234567,
    rate: 3,
    title: '__TITLE_TWO__',
    content: '__CONTENT_TWO__',
    likes: 12,
    dislikes: 6,
  },
  {
    id: 12345678,
    rate: 1,
    title: '__TITLE_THREE__',
    content: '__CONTENT_THREE__',
    likes: 2,
    dislikes: 4,
  },
]

const ratingAverage = 3.8
const sumRatingsOpinions = 4
const ratingLevels = {
  five_star: 5,
  four_star: 3,
  one_star: 9,
  three_star: 7,
  two_star: 0,
}

describe('ReviewProduct', () => {
  it.skip('should render and match snapshot with required props', () => {
    const tree = renderer
      .create(
        <ReviewProduct
          allReviews={allReviews}
          ratingAverage={ratingAverage}
          sumRatingsOpinions={sumRatingsOpinions}
          ratingLevels={ratingLevels}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    render(
      <ReviewProduct
        allReviews={allReviews}
        ratingAverage={ratingAverage}
        sumRatingsOpinions={sumRatingsOpinions}
        ratingLevels={ratingLevels}
      />
    )
    expect(screen.getByText('Todas')).toBeInTheDocument()
    expect(screen.getByText('Positivas')).toBeInTheDocument()
    expect(screen.getByText('Negativas')).toBeInTheDocument()
  })
})
