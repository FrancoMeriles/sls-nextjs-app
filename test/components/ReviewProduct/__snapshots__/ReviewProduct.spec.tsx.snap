// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewProduct should render and match snapshot with required props 1`] = `
Array [
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />,
  <p
    className="chakra-text css-1hlj7v5"
  >
    Opiniones sobre el producto
  </p>,
  <div
    className="css-pss9jm"
  >
    <div
      className="css-139sjt4"
    >
      <p
        className="chakra-text css-9j8q24"
      >
        3.8
      </p>
      <div
        className="css-0"
      >
        <svg
          className="chakra-icon css-3ii9cw"
          focusable={false}
          paint="full"
          viewBox="0 0 10 10"
        >
          <path
            d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
            fill="currentColor"
          />
        </svg>
        <svg
          className="chakra-icon css-3ii9cw"
          focusable={false}
          paint="full"
          viewBox="0 0 10 10"
        >
          <path
            d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
            fill="currentColor"
          />
        </svg>
        <svg
          className="chakra-icon css-3ii9cw"
          focusable={false}
          paint="full"
          viewBox="0 0 10 10"
        >
          <path
            d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
            fill="currentColor"
          />
        </svg>
        <svg
          className="chakra-icon css-3ii9cw"
          focusable={false}
          paint="half"
          viewBox="0 0 10 10"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M5.256 8L2.131 9.648l.597-3.49L.2 3.684l3.494-.509L5.256 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
              fill="#e0e0e0"
            />
            <path
              d="M5.272 8.026L2.137 9.679l.6-3.502L.2 3.697l3.505-.51L5.272 0z"
              fill="currentColor"
            />
          </g>
        </svg>
        <svg
          className="chakra-icon css-3ii9cw"
          focusable={false}
          paint="empty"
          viewBox="0 0 10 10"
        >
          <path
            d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
            fill="#e0e0e0"
          />
        </svg>
      </div>
      <p
        className="chakra-text css-1daurj3"
      >
        Promedio entre 
        4
         opiniones
      </p>
    </div>
    <div
      className="css-1r4umpa"
    >
      <div
        className="chakra-stack css-hvy9jy"
      >
        <p
          className="chakra-text css-x8wdze"
        >
          5
           estrellas
        </p>
        <div
          className="css-1nbic85"
        >
          <div
            className="css-kdwx3d"
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow={0}
              className="css-1jy0d03"
              role="progressbar"
              style={
                Object {
                  "width": "0%",
                }
              }
            />
          </div>
        </div>
        <p
          className="chakra-text css-1foer2f"
        >
          0
        </p>
      </div>
      <div
        className="chakra-stack css-hvy9jy"
      >
        <p
          className="chakra-text css-x8wdze"
        >
          4
           estrellas
        </p>
        <div
          className="css-1nbic85"
        >
          <div
            className="css-kdwx3d"
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow={175}
              className="css-1jy0d03"
              role="progressbar"
              style={
                Object {
                  "width": "175%",
                }
              }
            />
          </div>
        </div>
        <p
          className="chakra-text css-1foer2f"
        >
          7
        </p>
      </div>
      <div
        className="chakra-stack css-hvy9jy"
      >
        <p
          className="chakra-text css-x8wdze"
        >
          3
           estrellas
        </p>
        <div
          className="css-1nbic85"
        >
          <div
            className="css-kdwx3d"
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow={225}
              className="css-1jy0d03"
              role="progressbar"
              style={
                Object {
                  "width": "225%",
                }
              }
            />
          </div>
        </div>
        <p
          className="chakra-text css-1foer2f"
        >
          9
        </p>
      </div>
      <div
        className="chakra-stack css-hvy9jy"
      >
        <p
          className="chakra-text css-x8wdze"
        >
          2
           estrellas
        </p>
        <div
          className="css-1nbic85"
        >
          <div
            className="css-kdwx3d"
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow={75}
              className="css-1jy0d03"
              role="progressbar"
              style={
                Object {
                  "width": "75%",
                }
              }
            />
          </div>
        </div>
        <p
          className="chakra-text css-1foer2f"
        >
          3
        </p>
      </div>
      <div
        className="chakra-stack css-hvy9jy"
      >
        <p
          className="chakra-text css-x8wdze"
        >
          1
           estrellas
        </p>
        <div
          className="css-1nbic85"
        >
          <div
            className="css-kdwx3d"
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow={125}
              className="css-1jy0d03"
              role="progressbar"
              style={
                Object {
                  "width": "125%",
                }
              }
            />
          </div>
        </div>
        <p
          className="chakra-text css-1foer2f"
        >
          5
        </p>
      </div>
    </div>
  </div>,
  <div
    className="chakra-tabs css-0"
  >
    <div
      aria-orientation="horizontal"
      className="chakra-tabs__tablist css-k008qs"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-controls="tabs-2157099029-1--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-5wl5ud"
        id="tabs-2157099029-1--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Todas
      </button>
      <button
        aria-controls="tabs-2157099029-1--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-5wl5ud"
        id="tabs-2157099029-1--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Positivas
      </button>
      <button
        aria-controls="tabs-2157099029-1--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-5wl5ud"
        id="tabs-2157099029-1--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Negativas
      </button>
    </div>
    <div
      className="chakra-tabs__tab-panels css-8atqhb"
    >
      <div
        aria-labelledby="tabs-2157099029-1--tab-0"
        className="chakra-tabs__tab-panel css-47dblg"
        hidden={false}
        id="tabs-2157099029-1--tabpanel-0"
        role="tabpanel"
        tabIndex={0}
      >
        <div
          className="css-1puzibz"
        >
          <div
            className="css-0"
          >
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="full"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="full"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="full"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="full"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="empty"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="#e0e0e0"
              />
            </svg>
          </div>
          <p
            className="chakra-text css-160q44m"
          >
            __TITLE__
          </p>
          <p
            className="chakra-text css-g5hezf"
          >
            __CONTENT__
          </p>
          <div
            className="css-1ivb442"
          >
            <svg
              className="chakra-icon css-r96wal"
              focusable={false}
              viewBox="0 0 17 17"
            >
              <path
                d="M6.15 5.464l.86-.334L9.576 0l1.017.509a3 3 0 0 1 1.473 3.719l-.286.777h2.8a2.4 2.4 0 0 1 2.243 3.256l-1.205 3.159a4.2 4.2 0 0 1-4.682 2.635l-3.546-.65H6.15v1.2H2v-10.2h4.15v1.06zm0 1.288v5.453h1.294l.108.01 3.6.66a3 3 0 0 0 3.343-1.883L15.7 7.833a1.2 1.2 0 0 0-1.12-1.628h-4.52l.88-2.392a1.8 1.8 0 0 0-.829-2.202l-2.234 4.47-1.728.67zM3.2 13.404h1.75v-7.8H3.2v7.8z"
                fill="currentColor"
                fillRule="nonzero"
              />
            </svg>
             
            <p
              className="chakra-text css-g5hezf"
            >
              3
            </p>
            <svg
              className="chakra-icon css-10m0rwy"
              focusable={false}
              viewBox="0 0 17 17"
            >
              <path
                d="M6.15 5.464l.86-.334L9.576 0l1.017.509a3 3 0 0 1 1.473 3.719l-.286.777h2.8a2.4 2.4 0 0 1 2.243 3.256l-1.205 3.159a4.2 4.2 0 0 1-4.682 2.635l-3.546-.65H6.15v1.2H2v-10.2h4.15v1.06zm0 1.288v5.453h1.294l.108.01 3.6.66a3 3 0 0 0 3.343-1.883L15.7 7.833a1.2 1.2 0 0 0-1.12-1.628h-4.52l.88-2.392a1.8 1.8 0 0 0-.829-2.202l-2.234 4.47-1.728.67zM3.2 13.404h1.75v-7.8H3.2v7.8z"
                fill="currentColor"
                fillRule="nonzero"
              />
            </svg>
            <p
              className="chakra-text css-g5hezf"
            >
              1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="tabs-2157099029-1--tab-1"
        className="chakra-tabs__tab-panel css-47dblg"
        hidden={true}
        id="tabs-2157099029-1--tabpanel-1"
        role="tabpanel"
        tabIndex={0}
      >
        <div
          className="css-1puzibz"
        >
          <div
            className="css-0"
          >
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="full"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="full"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="full"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="full"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1phmqac"
              focusable={false}
              paint="empty"
              viewBox="0 0 10 10"
            >
              <path
                d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
                fill="#e0e0e0"
              />
            </svg>
          </div>
          <p
            className="chakra-text css-160q44m"
          >
            __TITLE__
          </p>
          <p
            className="chakra-text css-g5hezf"
          >
            __CONTENT__
          </p>
          <div
            className="css-1ivb442"
          >
            <svg
              className="chakra-icon css-r96wal"
              focusable={false}
              viewBox="0 0 17 17"
            >
              <path
                d="M6.15 5.464l.86-.334L9.576 0l1.017.509a3 3 0 0 1 1.473 3.719l-.286.777h2.8a2.4 2.4 0 0 1 2.243 3.256l-1.205 3.159a4.2 4.2 0 0 1-4.682 2.635l-3.546-.65H6.15v1.2H2v-10.2h4.15v1.06zm0 1.288v5.453h1.294l.108.01 3.6.66a3 3 0 0 0 3.343-1.883L15.7 7.833a1.2 1.2 0 0 0-1.12-1.628h-4.52l.88-2.392a1.8 1.8 0 0 0-.829-2.202l-2.234 4.47-1.728.67zM3.2 13.404h1.75v-7.8H3.2v7.8z"
                fill="currentColor"
                fillRule="nonzero"
              />
            </svg>
             
            <p
              className="chakra-text css-g5hezf"
            >
              3
            </p>
            <svg
              className="chakra-icon css-10m0rwy"
              focusable={false}
              viewBox="0 0 17 17"
            >
              <path
                d="M6.15 5.464l.86-.334L9.576 0l1.017.509a3 3 0 0 1 1.473 3.719l-.286.777h2.8a2.4 2.4 0 0 1 2.243 3.256l-1.205 3.159a4.2 4.2 0 0 1-4.682 2.635l-3.546-.65H6.15v1.2H2v-10.2h4.15v1.06zm0 1.288v5.453h1.294l.108.01 3.6.66a3 3 0 0 0 3.343-1.883L15.7 7.833a1.2 1.2 0 0 0-1.12-1.628h-4.52l.88-2.392a1.8 1.8 0 0 0-.829-2.202l-2.234 4.47-1.728.67zM3.2 13.404h1.75v-7.8H3.2v7.8z"
                fill="currentColor"
                fillRule="nonzero"
              />
            </svg>
            <p
              className="chakra-text css-g5hezf"
            >
              1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="tabs-2157099029-1--tab-2"
        className="chakra-tabs__tab-panel css-47dblg"
        hidden={true}
        id="tabs-2157099029-1--tabpanel-2"
        role="tabpanel"
        tabIndex={0}
      />
    </div>
  </div>,
]
`;
